website/src/components/SearchPage/DownloadDialog/DownloadUrlGenerator.ts

```diff
@@ -0,0 +1,134 @@
+        const baseUrl = `${this.lapisUrl}${getEndpoint(option.dataType)}`;
+        const params = new URLSearchParams();
+
+        params.set('downloadAsFile', 'true');
+        params.set('downloadFileBasename', this.generateFilename(option.dataType.type));
+        if (!option.includeOldData) {
+            params.set(VERSION_STATUS_FIELD, versionStatuses.latestVersion);
+            params.set(IS_REVOCATION_FIELD, 'false');
+        }
+        if (!option.includeRestricted) {
+            params.set('dataUseTerms', 'OPEN');
+        }
+        if (option.dataType.type === 'metadata') {
+            params.set('dataFormat', metadataDefaultDownloadDataFormat);
+        }
+        if (option.compression !== undefined) {
+            params.set('compression', option.compression);
+        }
+
+        switch (downloadParameters.type) {
+            case 'filter':
+                const lapisSearchParameters = downloadParameters.lapisSearchParameters;
+                if (lapisSearchParameters.accession !== undefined) {
+                    for (const accession of lapisSearchParameters.accession) {
+                        params.append('accession', accession);
+                    }
+                }
+
+                const mutationKeys = [
+                    'nucleotideMutations',
+                    'aminoAcidMutations',
+                    'nucleotideInsertions',
+                    'aminoAcidInsertions',
+                ];
+
+                for (const [key, value] of Object.entries(lapisSearchParameters)) {
+                    // Skip accession and mutations
+                    if (key === 'accession' || mutationKeys.includes(key)) {
+                        continue;
+                    }
+                    const stringValue = String(value);
+                    const trimmedValue = stringValue.trim();
+                    if (trimmedValue.length > 0) {
+                        params.set(key, trimmedValue);
+                    }
+                }
+
+                mutationKeys.forEach((key) => {
+                    if (lapisSearchParameters[key] !== undefined) {
+                        params.set(key, lapisSearchParameters[key].join(','));
+                    }
+                });
+                break;
+            case 'select':
+                const sortedIds = Array.from(downloadParameters.selectedSequences).sort();
+                sortedIds.forEach((accessionVersion) => {
+                    params.append('accessionVersion', accessionVersion);
+                });
+                break;
+        }
+
+        return {
+            url: `${baseUrl}?${params}`,
+            baseUrl,
+            params,
+        };
+    }
+
+    private generateFilename(dataType: string): string {
+        const timestamp = new Date().toISOString().slice(0, 16).replace('T', '-').replace(':', '');
+        return `${this.websiteName}_${this.organism}_${dataType}_${timestamp}`;
```

Comment:
```suggestion
        return `${this.websiteName.to_kebap()}_${this.organism.to_kebap()}_${dataType.to_kebap()}_${timestamp}`;
```

I think it's good to only use lower case in the file names, and so turn anything that might be camelCase into kebab-case as well.

Reply:
https://github.com/loculus-project/loculus/pull/3021/commits/8bebb799b732029c21509f405de80f66362ae5f8